daft/dataframe/dataframe.py

```diff
@@ -198,8 +198,9 @@ def iter_partitions(self) -> Iterator[Union[Table, "RayObjectRef"]]:
         else:
             # Execute the dataframe in a streaming fashion.
             context = get_context()
-            partitions_iter = context.runner().run_iter(self._builder)
-            yield from partitions_iter
+            results_iter = context.runner().run_iter(self._builder)
```

Comment:
I think you also may have to handle the `if self._result is not None:` case. we can add a test that does
```
df = df.collect()
for _ in df.iter_partitions():
  pass
```

Reply:
```
        if self._result is not None:
            # If the dataframe has already finished executing,
            # use the precomputed results.
            yield from self._result.values()
```

This block is still safe, because `PartitionSet.values()` still returns partitions (not `MaterializedResult`)
<img width="333" alt="image" src="https://github.com/Eventual-Inc/Daft/assets/17691182/09a85eb7-86b4-41e8-8063-5a546c772ae7">